socs/Lakeshore/Lakeshore370.py

```diff
@@ -0,0 +1,1783 @@
+                                           'attempts. Check connection.')
+                else:
+                    break
+        else:
+            self.com.write(msg_str)
+            resp = ''
+
+        if 'RDG' in message:
+            time.sleep(0.1)  # Instrument sampling rate of 10 readings/s max (pg 2-8)
+        else:
+            time.sleep(0.1)  # No comms for 100ms after sending message (manual says50ms)
+
+        #for line in traceback.format_stack():
+        #    print(line.strip())
+
+        print(msg_str)
+        print(resp)
+        return resp
+
+    def get_id(self):
+        """Get the ID number of the Lakeshore unit."""
+        return self.msg('*IDN?')
+
+    def get_temp(self, unit='kelvin', chan=-1):
+        """Get temperature from the Lakeshore.
+
+        Args:
+            unit (str): Unit to return reading for ('ohms' or 'kelvin')
+            chan (int): Channel to query, -1 for currently active channel
+
+        Returns:
+            float: The reading from the lakeshore, either in ohms or kelvin.
+
+        """
+        if (chan == -1):
+            resp = self.msg("SCAN?")
+            c = int(resp.split(',')[0])
+        else:
+            c = chan
+
+        channel = self.chan_num2Channel(c)
+
+        assert unit.lower() in ['ohms', 'kelvin']
+
+        if unit == 'ohms':
+            return float(channel.get_resistance_reading())
+        if unit == 'kelvin':
+            return float(channel.get_kelvin_reading())
+
+    def get_autoscan(self):
+        """Determine state of autoscan.
+
+        :returns: state of autoscanner
+        :rtype: bool
+        """
+        resp = self.msg('SCAN?')
+        scan_state = bool(int(resp.split(',')[1]))
+        self.autoscan = scan_state
+        return scan_state
+
+    def _set_autoscan(self, start=1, autoscan=0):
+        """Set the autoscan state and start channel for scanning.
+
+        :param start: Channel number to start scanning
+        :type start: int
+        :param autoscan: State of autoscan, 0 for off, 1 for on
+        :type autoscan: int
+        """
+        assert autoscan in [0, 1]
+
+        self.msg('SCAN {},{}'.format(start, autoscan))
+        self.autoscan = bool(autoscan)
+
+    def enable_autoscan(self):
+        """Enable the autoscan feature of the Lakeshore 372.
+
+        Will query active channel to pass already selected channel to SCAN
+        command.
+        """
+        active_channel = self.get_active_channel()
+        self.msg('SCAN {},{}'.format(active_channel.channel_num, 1))
+        self.autoscan = True
+
+    def disable_autoscan(self):
+        """Disable the autoscan feature of the Lakeshore 372.
+
+        Will query active channel to pass already selected channel to SCAN
+        command.
+        """
+        active_channel = self.get_active_channel()
+        self.msg('SCAN {},{}'.format(active_channel.channel_num, 0))
+        self.autoscan = False
+
+    def chan_num2Channel(self, channel_number):
```

Comment:
It is because there is no "control channel" to conveniently assign to index 0. This accounts for an the usual "off by one error" then, but would also handle more complex cases, such as if the LS370 constructor were modified to allow instantiation of an arbitrary set of channels.

Changed function name to chan_num2channel.